arangod/Cluster/ClusterInfo.cpp

```diff
@@ -949,6 +1009,72 @@ ClusterInfo::CollectionWithHash ClusterInfo::buildCollection(
+              if (vocbaseViews == _newPlannedViews.end() ||
+                  vocbaseViews->second.find(viewId) ==
+                      vocbaseViews->second.end()) {
+                if (_pendingCleanups.find(idx->id()) ==
+                    _pendingCleanups.end()) {
+                  // scheduling low prio index drop
+                  auto* scheduler = SchedulerFeature::SCHEDULER;
+                  if (scheduler && !_server.isStopping()) {
+                    LOG_TOPIC("0d7b2", WARN, Logger::CLUSTER)
+                        << "Scheduling drop for dangling link " << idx->id()
+                        << " View: " << viewId << " was dropped";
+                    _currentCleanups.insert(idx->id());
+                    scheduler->queue(
+                        RequestLane::INTERNAL_LOW,
+                        [id = idx->id(), name = collection->name(),
+                         vocbase = vocbase.name(), &server = server()] {
+                          if (!server.isStopping()) {
+                            auto& ci = server.getFeature<ClusterFeature>()
+                                           .clusterInfo();
+                            auto coll = ci.getCollectionNT(vocbase, name);
+                            if (coll) {
+                              velocypack::Builder builder;
+                              builder.openObject();
+                              builder.add(arangodb::StaticStrings::IndexId,
+                                          velocypack::Value(id.id()));
+                              builder.close();
+                              LOG_TOPIC("d7665", TRACE, Logger::CLUSTER)
+                                  << "Dropping dangling link " << id;
+                              auto res = methods::Indexes::drop(
+                                  coll.get(), builder.slice());
+                              if (res.fail() &&
+                                  res.errorNumber() !=
+                                      TRI_ERROR_ARANGO_INDEX_NOT_FOUND) {
+                                LOG_TOPIC("b27f3", WARN, Logger::CLUSTER)
+                                    << "Failed to drop dangling link " << id
+                                    << " Err:" << res.errorNumber();
+                                ci.unregisterDangling(id);
+                              } else {
+                                LOG_TOPIC("2c47a", TRACE, Logger::CLUSTER)
+                                    << "Removed dangling link" << id;
+                              }
+                            } else {
+                              LOG_TOPIC("f5596", TRACE, Logger::CLUSTER)
+                                  << "Scheduled drop for dangling link " << id
+                                  << " skipped as collection is dropped";
+                            }
+                          }
+                        });
+                  }
+                } else {
```

Comment:
Because it's written in big block I don't understand to which block this else belong.

In general it's not clear when you insert to _currentCleanups and when not